components/feature/pwa/src/main/java/mozilla/components/feature/pwa/ManifestStorage.kt

```diff
@@ -31,6 +35,47 @@ class ManifestStorage(context: Context) {
         manifestDao.value.getManifest(startUrl)?.manifest
     }
 
+    /**
+     * Load all Web App Manifests with a matching scope for the given URL from disk.
+     * If no manifest is found, an empty list is returned.
+     *
+     * @param url URL of site. Should correspond to an url covered by the scope of a stored manifest.
+     * @param includeTimeout Whether the manifest should only be loaded if the last usage
+     *                       has not passed the defined timeout.
+     * @param currentTime timestamp in milliseconds that is used to validate timeout.
+     */
+    suspend fun loadManifestsByScope(
+        url: String,
+        includeTimeout: Boolean = false,
+        currentTime: Long = System.currentTimeMillis()
+    ): List<WebAppManifest> = withContext(Dispatchers.IO) {
+        val entityList = manifestDao.value.getManifestsByScope(url)
+        val first = entityList.firstOrNull() ?: return@withContext emptyList<WebAppManifest>()
+
+        entityList.filter { it ->
+            if (it.scope != first.scope) {
```

Comment:
I think we should either return the entire list even if some manifests have a narrower scope, or always return only 1 item instead of the list. The scope check here seems arbitrary to me.

Reply:
I agree, let's just return the entire list. Since it's always sorted by narrowest scope first, I think it also will be simple to get the best match by just taking the first entry.